iothub/device/src/IotHubTokenRefresher.cs

```diff
@@ -10,68 +10,62 @@ namespace Microsoft.Azure.Devices.Client
     using Microsoft.Azure.Amqp;
     using System.Diagnostics;
     using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Azure.Devices.Client.Transport;
 
-    internal sealed class IotHubTokenRefresher
+    internal sealed class AmqpTokenRefresher : IDisposable
     {
         private static readonly string[] AccessRightsStringArray = 
             AccessRightsHelper.AccessRightsToStringArray(AccessRights.DeviceConnect);
 
         static readonly TimeSpan BufferPeriod = TimeSpan.FromSeconds(120);
 
-        private readonly AmqpSession amqpSession;
+        private readonly AmqpClientSession amqpClientSession;
         private readonly IotHubConnectionString connectionString;
         private readonly string audience;
         private readonly CancellationTokenSource cancellationTokenSource;
         private volatile bool taskCancelled;
 
-        public IotHubTokenRefresher(AmqpSession amqpSession, IotHubConnectionString connectionString, string audience)
+        public AmqpTokenRefresher(AmqpClientSession amqpClientSession, IotHubConnectionString connectionString, string audience)
         {
-            this.amqpSession = amqpSession ?? throw new ArgumentNullException("amqpSession");
+            this.amqpClientSession = amqpClientSession ?? throw new ArgumentNullException("amqpClientSession");
             this.connectionString = connectionString;
             this.audience = audience;
             this.cancellationTokenSource = new CancellationTokenSource();
         }
 
-        public void Cancel()
```

Comment:
I understand internal should be the right one...but since it was previously "public" API.  Changing them now would be considered breaking change?

Reply:
We will discuss this this morning. My opinion: the whole redesign is breaking change , so this the right time to do it right (e.g. reduce the scope)

---
In reply to: [251634408](https://github.com/Azure/azure-iot-sdk-csharp/pull/770#discussion_r251634408) [](ancestors = 251634408)

Reply:
We discussed and created a task to bring back the removed classes and deprecate them. Also we agreed to do it right this time and use the lower visibility in internal classes

---
In reply to: [251931830](https://github.com/Azure/azure-iot-sdk-csharp/pull/770#discussion_r251931830) [](ancestors = 251931830,251634408)